supabase-openai-summary/src/steps/importers.py

```diff
@@ -0,0 +1,57 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import List
+from datetime import datetime, timedelta, timezone
+
+from zenml.steps import BaseParameters, step
+from supabase import create_client, Client as SupabaseClient
+from zenml.client import Client
+
+
+class SupabaseReaderParams(BaseParameters):
+    table_name: str = "analytics"
+    filter_date_column: str = "created_at"  # assume supabase timestampz col
+    filter_interval_hours: int = 24  # interval for hours
+    summary_column: str = "video_title"  # the column to summarize
+    limit: int = 300  # limit the number of rows to read
+
+
+@step(enable_cache=False)
+def supabase_reader(
```

Comment:
Cool. Perhaps we should submit this as a Hub plugin even?

Reply:
Lets do that in the time we work on the hub, might need to more generic